tools/cluster_mgr.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+import argparse
+import json
+import math
+import redis
+import subprocess
+import time
+
+'''
+To install: pip install -r requirements.txt
+'''
+
+
+class Node:
+    def __init__(self, port, admin_port):
+        self.port = port
+        self.admin_port = admin_port
+
+    id = ''
+
+
+def start_node(node, threads):
+    f = open(f'/tmp/dfly.cluster.node.{node.port}.log', 'w')
+    print(f'- Log file for node {node.port}: {f.name}')
+    subprocess.Popen(['../build-dbg/dragonfly', f'--port={node.port}',
+                      f'--admin_port={node.admin_port}', '--cluster_mode=yes', f'--proactor_threads={threads}',
+                      '--dbfilename=', f'--logtostderr'], stderr=f)
+
+
+def send_command(node, command):
+    for i in range(0, 5):
+        try:
+            client = redis.Redis(decode_responses=True,
+                                 host="localhost", port=node.admin_port)
+            result = client.execute_command(*command)
+            client.close()
+            return result
+        except:
+            time.sleep(0.1)
+
+    print(
+        f'Unable to connect to localhost:{node.admin_port} after 5 attempts!')
+
+
+def update_id(node):
+    id = send_command(node, ['dflycluster', 'myid'])
+    node.id = id
+    print(f'- ID for {node.port}: {id}')
+
+
+def build_config(masters, replicas):
+    slots_per_node = 16384 / len(masters)
+
+    def build_node(node):
+        return {
+            "id": node.id,
+            "ip": "localhost",
+            "port": node.port
+        }
+
+    config = []
+    for i, master in enumerate(masters):
+        c = {
+            "slot_ranges": [
+                {
+                    "start": math.floor(i * slots_per_node),
+                    "end": math.floor((i+1) * slots_per_node - 1)
+                }
+            ],
+            "master": build_node(master),
+            "replicas": []
+        }
+
+        if len(replicas) > 0:
+            c["replicas"].append(build_node(replicas[i]))
+
+        config.append(c)
+
+    return json.dumps(config, indent=2)
+
+
+def push_config(nodes, config):
+    for node in nodes:
+        response = send_command(node, ['dflycluster', 'config', config])
+        print(f'- Push into {node.port}: {response}')
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Local Cluster Manager')
+    parser.add_argument('--num_masters', type=int, default=3,
```

Comment:
could be cool to support "down" operation as well. In case we start 30 node cluster - could be cumbersome to take it down afterwards.

Reply:
I just run it via `killall dragonfly; ./cluster_mgr.py` :)
But yeah, I plan to extend the functionality here further. First I'll likely implement slot reallocation - so the script will connect to an existing cluster, get config and push new one

Reply:
also I would rename `num_masters` to `shards` or `num_shards`

Reply:
I actually think this is more confusing. I wouldn't know without reading if shards include replicas or not, and it's also re-using the term shards from the C++ code (like `EngineShard` etc). If you feel strongly about it I'll obviously rename, but otherwise perhaps it's better to keep as is.